security/yubi-key.md

```diff
@@ -39,52 +39,70 @@ Same as in OTP case, you will need to setup your YubiKey, choose separate
+       
+2. Configure your YubiKey for challenge-reponse `HMAC-SHA1` mode, for example
    [following this
-   tutorial](https://www.yubico.com/products/services-software/personalization-tools/challenge-response/)
-2. Install `ykpers` package in template on which your USB VM is based.
-3. Create `/usr/local/bin/yubikey-auth` script:
+   tutorial](https://www.yubico.com/products/services-software/personalization-tools/challenge-response/).
+   
+   On Debian, you can run the graphical user interface `yubikey-personalization-gui` from the command line.
+   
+   - Choose `configuration slot 2`.
+   - It is recommended to enable `Require user input (button press)` but this is optional.
+   - Note: Derivating from the above video, use the following settings select
+   `HMAC-SHA1 mode`: `fixed 64 bit input`. 
+   - We will refer the `Secret Key (20 bytes hex)` as `AESKEY`. It is recommended to keep a backup of
+   your `AESKEY` in an offline VM used as vault. In case you have multiple yubikeys for backup purposes
+   (in case a yubikey gets lost, stolen or breaks) you can write the same settings into other yubikeys.
 
-       #!/bin/sh
+3. Install [qubes-app-yubikey](https://github.com/QubesOS/qubes-app-yubikey) in dom0.
 
-       key="$1"
+       sudo qubes-dom0-update qubes-dom0-yubikey
 
-       if [ -z "$key" ]; then
-           echo "Usage: $0 <AESKEY> [<PASSWORD-HASH>]"
-           exit 1
-       fi
+4. Adjust USB VM name in case you are using something other than the default
+   `sys-usb` by editing `/etc/qubes/yk-keys/yk-vm` in dom0.
 
-       # if password has given, verify it
-       if [ -n "$2" ]; then
-           # PAM appends \0 at the end
-           hash=`head -c -1 | openssl dgst -sha1 -r | cut -f1 -d ' '`
-           if [ "x$2" != "x$hash" ]; then
-               exit 1
-           fi
-       fi
+5. Paste your `AESKEY` from step 2 into `/etc/qubes/yk-keys/yk-secret-key.hex` in dom0.
 
-       challenge=`head -c64 /dev/urandom | xxd -c 64 -ps`
-       # You may need to adjust slot number and USB VM name here
-       response=`qvm-run -u root --nogui -p sys-usb "ykchalresp -2 -x $challenge"`
+6. Paste your hashed password (other than your standard Qubes password)  into
+`/etc/qubes/yk-keys/yk-login-pass-hashed.hex` in dom0.
 
-       correct_response=`echo $challenge | xxd -r -ps | openssl dgst -sha1 -macopt hexkey:$key -mac HMAC -r | cut -f1 -d ' '`
+   You can calculate your hashed password using this command:
 
-       test "x$correct_response" = "x$response"
-       exit $?
+       echo -n "PASSWORD" | openssl dgst -sha1
+       
+   (Replace `PASSWORD` with your actual password.)
 
-4. Adjust USB VM name (`sys-usb` above), and possibly YubiKey slot number (`2`
-   above), then make the script executable.
-5. Edit `/etc/pam.d/xscreensaver` (or appropriate file if you are using other
-   screen locker program). Add this line at the beginning:
+7. Create `/etc/pam.d/yubikey` in dom0 with the following content:
```

Comment:
Since you've added it to the package, this step isn't needed anymore.

Reply:
Didn't know it would propagate so quickly! Awesome!

Fixed.